linux/src/socket_transport_linux.c

```diff
@@ -619,15 +620,15 @@ SOCKET_TRANSPORT_HANDLE socket_transport_accept(SOCKET_TRANSPORT_HANDLE socket_t
                 if (accepted_socket == INVALID_SOCKET)
```

Comment:
>if (accepted_socket == INVALID_SOCKET) [](http://example.com/codeflow?start=16&length=38)

You need to check here if errno is set to "EAGAIN" or "EWOULDBLOCK", this would mean that you need to return SOCKET_ACCEPT_NO_CONNECTION instead of just SOCKET_ACCEPT_ERROR

Here is the doc for linux [https://linux.die.net/man/2/accept](https://linux.die.net/man/2/accept) just fyi